app/src/test/java/com/ikarabulut/BoardTest.java

```diff
@@ -87,6 +87,91 @@ void setAliveAt_When1_1IsPassed_ReturnsTheCellObjectAt1_1() {
 
     }
 
+    @Nested
+    @DisplayName("willCellSurviveAt()")
+    class WillCellSurviveAt {
+        private void setDeadBoard() {
+            for ( ArrayList<Cell> boardRow : board.getBoard()) {
+                for ( Cell cell : boardRow) {
+                    cell.die();
+                }
+            }
+        }
+
+        public Board board;
+        @BeforeEach
+        void createBoard() {
+            board = new Board();
+        }
+
+        @Test
+        @DisplayName("Alive Cell with less than 2 alive neighbors should not be alive next generation")
+        void willCellSurviveAt_LessThan2() {
+            setDeadBoard();
+            board.setAliveAt(0, 1);
+            board.setAliveAt(0, 0);
+
+            assertFalse(board.willSurviveAt(0, 1));
+        }
+
+        @Test
+        @DisplayName("Alive Cell with 2 alive neighbors should be alive next generation")
+        void willCellSurviveAt_With2() {
+            setDeadBoard();
+            board.setAliveAt(0, 1);
+            board.setAliveAt(0, 0);
+            board.setAliveAt(1, 0);
+
+            assertTrue(board.willSurviveAt(0, 1));
+        }
+
+        @Test
+        @DisplayName("Alive Cell with 3 alive neighbors should be alive next generation")
+        void willCellSurviveAt_With3() {
+            setDeadBoard();
+            board.setAliveAt(0, 1);
+            board.setAliveAt(0, 0);
+            board.setAliveAt(1, 0);
+            board.setAliveAt(0, 2);
+
+            assertTrue(board.willSurviveAt(0, 1));
+        }
+
+        @Test
+        @DisplayName("Alive Cell with 4 alive neighbors should be dead next generation")
+        void willCellSurviveAt_With4() {
+            setDeadBoard();
+            board.setAliveAt(0, 1);
+            board.setAliveAt(0, 0);
+            board.setAliveAt(1, 0);
+            board.setAliveAt(0, 2);
+            board.setAliveAt(1, 2);
+
+            assertFalse(board.willSurviveAt(0, 1));
+        }
+
+        @Test
+        @DisplayName("Dead Cell with 3 alive neighbors should be alive next generation")
+        void willCellSurviveAt_DeadWith3() {
+            setDeadBoard();
+            board.setAliveAt(0, 0);
+            board.setAliveAt(1, 0);
+            board.setAliveAt(0, 2);
+
+            assertTrue(board.willSurviveAt(0, 1));
+        }
+
+        @Test
+        @DisplayName("Dead Cell with anything other than 3 alive neighbors should be dead next generation")
```

Comment:
This test name is a little misleading.  I'm not sure what "anything other than..." means.  It sounds like an exhaustive test. But at face value it seems to be duplicating the test in `willCellSurviveAt_LessThan2`.

Am I misunderstanding?

Reply:
You have it correct. I was uneasy with this test name, because it is really only testing one case of alive cells which is less than 2. I was planning to add a few different assertions but ended up taking the easy way out with this one. Ill update this name